### ISSUE TITLE
Crash in VMWCoreBackdoor::BackdoorRPCSend

### ISSUE THREAD

**Issue by diversys** [HaikuArchives/VMwareAddons#14]
VMware® Workstation 14 Pro
14.1.1 build-7528167

Shrink virtuall disks->Shrink now

```
Debug information for team /boot/system/Deskbar (45600):
CPU(s): 2x Intel Core™ i5-7200U
Memory: 1.49 GiB total, 583.38 MiB used
Haiku revision: hrev51858 Mar 24 2018 14:07:08 (x86_64)

Active Threads:
	thread 45600: Deskbar (main)
	thread 45605: _BMediaRoster_ 
	thread 45606: Expando Window Watcher 
	thread 45609: w>Twitcher 
	thread 45610: w>Deskbar 
	thread 46500: team 45600 debug task 
	thread 46495: vmw cleanup 
		state: Exception (Segment violation)

		Frame		IP			Function Name
		-----------------------------------------------
		0x000010	0x5b340b55cb	VMWCoreBackdoor::BackdoorRPCSend(regs_t*, char*, unsigned long) + 0x3b 
			Disassembly:
				VMWCoreBackdoor::BackdoorRPCSend(regs_t*, char*, unsigned long):
				0x0000005b340b5590:               55  push %rbp
				0x0000005b340b5591:           4889e5  mov %rsp, %rbp
				0x0000005b340b5594:           4989f0  mov %rsi, %r8
				0x0000005b340b5597:           4889d6  mov %rdx, %rsi
				0x0000005b340b559a:         488b4718  mov 0x18(%rdi), %rax
				0x0000005b340b559e:           498900  mov %rax, (%r8)
				0x0000005b340b55a1:         49894810  mov %rcx, 0x10(%r8)
				0x0000005b340b55a5:         488b5710  mov 0x10(%rdi), %rdx
				0x0000005b340b55a9:   4881ca59560000  or $0x5659, %rdx
				0x0000005b340b55b0:         49895018  mov %rdx, 0x18(%r8)
				0x0000005b340b55b4:         49897020  mov %rsi, 0x20(%r8)
				0x0000005b340b55b8:         488b7f20  mov 0x20(%rdi), %rdi
				0x0000005b340b55bc:               53  push %rbx
				0x0000005b340b55bd:               55  push %rbp
				0x0000005b340b55be:             89c5  mov %eax, %ebp
				0x0000005b340b55c0:       b868584d56  mov $0x564d5868, %eax
				0x0000005b340b55c5:       bb00000100  mov $0x10000, %ebx
				0x0000005b340b55ca:               fc  cld 
				0x0000005b340b55cb:           67f36e  a32 rep outsb  <--

		00000000	00000000	? 

		Registers:
			  rip:	0x0000005b340b55cb
			  rsp:	0x00007f461139e670
			  rbp:	0x0000000000000000
			  rax:	0x00000000564d5868
			  rbx:	0x0000000000010000
			  rcx:	0x000000000000000b
			  rdx:	0x0000000000005659
			  rsi:	0x0000005b340b6306
			  rdi:	0x0000000000000000
			   r8:	0x00007f461139e690
			   r9:	0x000000000001001e
			  r10:	0x0000000000000000
			  r11:	0x0000000000003246
			  r12:	0x0000000018045f58
			  r13:	0x0000000000000000
			  r14:	0x0000005b340b6306
			  r15:	0x0000000000000001
			   cs:	0x0023
			   ds:	0x0000
			   es:	0x0000
			   fs:	0x0000
			   gs:	0x0000
			   ss:	0x001b
			  st0:	0
			  st1:	0
			  st2:	0
			  st3:	0
			  st4:	0
			  st5:	0
			  st6:	3.03e+04
			  st7:	6.41e+07
			  mm0:	{0, 0, 0, 0}
			  mm1:	{0, 0, 0, 0}
			  mm2:	{0, 0, 0, 0}
			  mm3:	{0, 0, 0, 0}
			  mm4:	{0, 0, 0, 0}
			  mm5:	{0, 0, 0, 0}
			  mm6:	{0, 0, 0, 0xecba}
			  mm7:	{0x9e0b, 0x103c, 0xdfb2, 0xf45f}
			 xmm0:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm1:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm2:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm3:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm4:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm5:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm6:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm7:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm8:	{0, 0, 0, 0, 0, 0, 0, 0}
			 xmm9:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm10:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm11:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm12:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm13:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm14:	{0, 0, 0, 0, 0, 0, 0, 0}
			xmm15:	{0, 0, 0, 0, 0, 0, 0, 0}

Loaded Images:
	ID		Text Base	Text End	Data Base	Data End	Type	Name
	--------------------------------------------------------------------------------
	1051046	0xcb11d0000	0xcb1289000	0xcb1489000	0xcb1496000	lib    	/boot/system/lib/libmedia.so
	1051045	0xcdded3000	0xcddee8000	0xcde0e7000	0xcde0e9000	add-on 	/boot/system/bin/desklink
	1051021	0xe92c3d000	0xe92ca4000	0xe92ea3000	0xe92eab000	app    	/boot/system/Deskbar
	1051044	0x455c1d5000	0x455c240000	0x455c43f000	0x455c44a000	lib    	/boot/system/lib/libssl.so.1.0.0
	1051222	0x5b340a4000	0x5b340b9000	0x5b342b8000	0x5b342ba000	add-on 	/boot/system/bin/vmware_tray
	1051032	0x5b8a6ba000	0x5b8a6d6000	0x5b8a8d5000	0x5b8a8d6000	lib    	/boot/system/lib/libz.so.1.2.11
	1051031	0x65276f0000	0x6527888000	0x6527a87000	0x6527a9c000	lib    	/boot/system/lib/libicuuc.so.57.1
	1051024	0x6fa57a3000	0x6fa5905000	0x6fa5b05000	0x6fa5b19000	lib    	/boot/system/lib/libstdc++.so.6.0.21
	1051027	0x93d8188000	0x93d8194000	0x93d8393000	0x93d8395000	lib    	/boot/system/lib/libicuio.so.57.1
	1051043	0xae58b28000	0xae58d34000	0xae58f34000	0xae58f5d000	lib    	/boot/system/lib/libcrypto.so.1.0.0
	1051040	0xbcbbb79000	0xbcbbbb8000	0xbcbbdb7000	0xbcbbdc0000	lib    	/boot/system/lib/libnetwork.so
	1051048	0xc833597000	0xc8335b7000	0xc8337b6000	0xc8337b9000	add-on 	/boot/system/preferences/KeymapSwitcher
	1051023	0xd1fd2ce000	0xd1fd466000	0xd1fd666000	0xd1fd67f000	lib    	/boot/system/lib/libtracker.so
	1051039	0xd9cf5a7000	0xd9cf5bb000	0xd9cf7ba000	0xd9cf7bc000	add-on 	/boot/system/apps/NetworkStatus
	1051034	0xe75932c000	0xe759341000	0xe759540000	0xe759542000	lib    	/boot/system/lib/libtranslation.so
	1051033	0xec91eea000	0xec91f95000	0xec92194000	0xec92195000	lib    	/boot/system/lib/libzstd.so.1.3.3
	1051025	0xfc372a4000	0xfc38b22000	0xfc38d21000	0xfc38d22000	lib    	/boot/system/lib/libicudata.so.57.1
	1051026	0x1182fb49000	0x1182fdc6000	0x1182ffc5000	0x1182ffda000	lib    	/boot/system/lib/libicui18n.so.57.1
	1051022	0x12aad07b000	0x12aad37e000	0x12aad57d000	0x12aad5ab000	lib    	/boot/system/lib/libbe.so
	1051038	0x12fec079000	0x12fec09c000	0x12fec29c000	0x12fec29f000	add-on 	/boot/system/apps/ProcessController
	1051018	0x13bdae36000	0x13bdae57000	0x13bdb056000	0x13bdb058000	lib    	/boot/system/runtime_loader
	1051041	0x14dc68a6000	0x14dc690a000	0x14dc6b09000	0x14dc6b0e000	lib    	/boot/system/lib/libbnetapi.so
	1051019	0x159835ea000	0x15983742000	0x15983941000	0x15983994000	lib    	/boot/system/lib/libroot_debug.so
	1051037	0x17ec718a000	0x17ec719b000	0x17ec739b000	0x17ec739c000	lib    	/boot/system/lib/libroot-addon-icu.so
	1051029	0x185244ed000	0x185244f8000	0x185246f8000	0x185246f9000	lib    	/boot/system/lib/libiculx.so.57.1
	1051020	0x1939a5cf000	0x1939a5e4000	0x1939a7e3000	0x1939a7e4000	lib    	/boot/system/lib/libgcc_s.so.1
	1051047	0x1a2c8e54000	0x1a2c8e62000	0x1a2c9062000	0x1a2c9063000	add-on 	/boot/system/apps/NetPulse
	1051030	0x1a2db63d000	0x1a2db66c000	0x1a2db86c000	0x1a2db8a0000	lib    	/boot/system/lib/libicutu.so.57.1
	1051042	0x1cf6b1c6000	0x1cf6b1cb000	0x1cf6b3ca000	0x1cf6b3cb000	lib    	/boot/system/lib/libbsd.so
	1051035	0x1f596d7d000	0x1f596e79000	0x1f597079000	0x1f59707c000	lib    	/boot/system/lib/libtextencoding.so
	1051028	0x1f6e6a28000	0x1f6e6a7d000	0x1f6e6c7d000	0x1f6e6c80000	lib    	/boot/system/lib/libicule.so.57.1
	1051017	0x7f5db6e6b000	0x7f5db6e73000	0x00000000	0x00000000	system 	commpage

Areas:
	ID		Base		End			Size (KiB)	Protection	Locking			Name
	--------------------------------------------------------------------------------
	3936508	0x0013c000	0x0033c000	      2048	rw-        	full          	additional heap area
	3935683	0x00497000	0x00697000	      2048	rw-        	full          	additional heap area
	3935667	0x010a0000	0x012a0000	      2048	rw-        	full          	additional heap area
	3935666	0x0202a000	0x02042000	        96	rw-        	full          	rld heap
	3935665	0x059cc000	0x059e4000	        96	rw-        	full          	rld heap
	3935664	0x093a3000	0x093b3000	        64	rw-        	full          	rld heap
	3935663	0x0e83a000	0x0e84a000	        64	rw-        	full          	rld heap
	3935629	0x18000000	0x18100000	      1024	rw-        	full          	heap
	3935676	0x60000000	0x60001000	         4	rw-S       	full          	server_memory
	3935678	0x68000000	0x6802c000	       176	rw-S       	full          	server_memory
	3936131	0x70000000	0x70020000	       128	rw-S       	full          	server_memory
	3935702	0xcb11d0000	0xcb1289000	       740	r-x        	full          	libmedia.so_seg0ro
	3935703	0xcb1489000	0xcb1495000	        48	rw-        	full          	libmedia.so_seg1rw
	3935704	0xcb1495000	0xcb1496000	         4	rw-        	full          	libmedia.so_seg2rw
	3935700	0xcdded3000	0xcddee8000	        84	r-x        	full          	desklink_seg0ro
	3935701	0xcde0e7000	0xcde0e9000	         8	rw-        	full          	desklink_seg1rw
	3935630	0xe92c3d000	0xe92ca4000	       412	r-x        	full          	Deskbar_seg0ro
	3935631	0xe92ea3000	0xe92eab000	        32	rw-        	full          	Deskbar_seg1rw
	3935623	0xf696d7000	0xf696d8000	         4	rw-S       	full          	_rld_debug_
	3935622	0x1ca1be6000	0x1ca1bf6000	        64	rw-        	full          	rld heap
	3935698	0x455c1d5000	0x455c240000	       428	r-x        	full          	libssl.so.1.0.0_seg0ro
	3935699	0x455c43f000	0x455c44a000	        44	rw-        	full          	libssl.so.1.0.0_seg1rw
	3936279	0x5b340a4000	0x5b340b9000	        84	r-x        	full          	vmware_tray_seg0ro
	3936280	0x5b342b8000	0x5b342ba000	         8	rw-        	full          	vmware_tray_seg1rw
	3935655	0x5b8a6ba000	0x5b8a6d6000	       112	r-x        	full          	libz.so.1.2.11_seg0ro
	3935656	0x5b8a8d5000	0x5b8a8d6000	         4	rw-        	full          	libz.so.1.2.11_seg1rw
	3935653	0x65276f0000	0x6527888000	      1632	r-x        	full          	libicuuc.so.57.1_seg0ro
	3935654	0x6527a87000	0x6527a9c000	        84	rw-        	full          	libicuuc.so.57.1_seg1rw
	3935637	0x6fa57a3000	0x6fa5905000	      1416	r-x        	full          	libstdc++.so.6.0.21_seg0ro
	3935638	0x6fa5b05000	0x6fa5b11000	        48	rw-        	full          	libstdc++.so.6.0.21_seg1rw
	3935639	0x6fa5b11000	0x6fa5b19000	        32	rw-        	full          	libstdc++.so.6.0.21_seg2rw
	3935644	0x93d8188000	0x93d8194000	        48	r-x        	full          	libicuio.so.57.1_seg0ro
	3935645	0x93d8393000	0x93d8395000	         8	rw-        	full          	libicuio.so.57.1_seg1rw
	3935695	0xae58b28000	0xae58d34000	      2096	r-x        	full          	libcrypto.so.1.0.0_seg0ro
	3935696	0xae58f34000	0xae58f5b000	       156	rw-        	full          	libcrypto.so.1.0.0_seg1rw
	3935697	0xae58f5b000	0xae58f5d000	         8	rw-        	full          	libcrypto.so.1.0.0_seg2rw
	3935688	0xbcbbb79000	0xbcbbbb8000	       252	r-x        	full          	libnetwork.so_seg0ro
	3935689	0xbcbbdb7000	0xbcbbdba000	        12	rw-        	full          	libnetwork.so_seg1rw
	3935690	0xbcbbdba000	0xbcbbdc0000	        24	rw-        	full          	libnetwork.so_seg2rw
	3935711	0xc833597000	0xc8335b7000	       128	r-x        	full          	KeymapSwitcher_seg0ro
	3935712	0xc8337b6000	0xc8337b9000	        12	rw-        	full          	KeymapSwitcher_seg1rw
	3935634	0xd1fd2ce000	0xd1fd466000	      1632	r-x        	full          	libtracker.so_seg0ro
	3935635	0xd1fd666000	0xd1fd67e000	        96	rw-        	full          	libtracker.so_seg1rw
	3935636	0xd1fd67e000	0xd1fd67f000	         4	rw-        	full          	libtracker.so_seg2rw
	3935686	0xd9cf5a7000	0xd9cf5bb000	        80	r-x        	full          	NetworkStatus_seg0ro
	3935687	0xd9cf7ba000	0xd9cf7bc000	         8	rw-        	full          	NetworkStatus_seg1rw
	3935659	0xe75932c000	0xe759341000	        84	r-x        	full          	libtranslation.so_seg0ro
	3935660	0xe759540000	0xe759542000	         8	rw-        	full          	libtranslation.so_seg1rw
	3935657	0xec91eea000	0xec91f95000	       684	r-x        	full          	libzstd.so.1.3.3_seg0ro
	3935658	0xec92194000	0xec92195000	         4	rw-        	full          	libzstd.so.1.3.3_seg1rw
	3935640	0xfc372a4000	0xfc38b22000	     25080	r-x        	full          	libicudata.so.57.1_seg0ro
	3935641	0xfc38d21000	0xfc38d22000	         4	rw-        	full          	libicudata.so.57.1_seg1rw
	3935642	0x1182fb49000	0x1182fdc6000	      2548	r-x        	full          	libicui18n.so.57.1_seg0ro
	3935643	0x1182ffc5000	0x1182ffda000	        84	rw-        	full          	libicui18n.so.57.1_seg1rw
	3935632	0x12aad07b000	0x12aad37e000	      3084	r-x        	full          	libbe.so_seg0ro
	3935633	0x12aad57d000	0x12aad5ab000	       184	rw-        	full          	libbe.so_seg1rw
	3935684	0x12fec079000	0x12fec09c000	       140	r-x        	full          	ProcessController_seg0ro
	3935685	0x12fec29c000	0x12fec29f000	        12	rw-        	full          	ProcessController_seg1rw
	3935620	0x13bdae36000	0x13bdae57000	       132	r-x        	full          	runtime_loader_seg0ro
	3935621	0x13bdb056000	0x13bdb058000	         8	rw-        	full          	runtime_loader_seg1rw
	3935691	0x14dc68a6000	0x14dc690a000	       400	r-x        	full          	libbnetapi.so_seg0ro
	3935692	0x14dc6b09000	0x14dc6b0e000	        20	rw-        	full          	libbnetapi.so_seg1rw
	3935624	0x159835ea000	0x15983742000	      1376	r-x        	full          	libroot_debug.so_seg0ro
	3935625	0x15983941000	0x1598394f000	        56	rw-        	full          	libroot_debug.so_seg1rw
	3935626	0x1598394f000	0x15983994000	       276	rw-        	full          	libroot_debug.so_seg2rw
	3935670	0x17ec718a000	0x17ec719b000	        68	r-x        	full          	libroot-addon-icu.so_seg0ro
	3935671	0x17ec739b000	0x17ec739c000	         4	rw-        	full          	libroot-addon-icu.so_seg1rw
	3935648	0x185244ed000	0x185244f8000	        44	r-x        	full          	libiculx.so.57.1_seg0ro
	3935649	0x185246f8000	0x185246f9000	         4	rw-        	full          	libiculx.so.57.1_seg1rw
	3935627	0x1939a5cf000	0x1939a5e4000	        84	r-x        	full          	libgcc_s.so.1_seg0ro
	3935628	0x1939a7e3000	0x1939a7e4000	         4	rw-        	full          	libgcc_s.so.1_seg1rw
	3935709	0x1a2c8e54000	0x1a2c8e62000	        56	r-x        	full          	NetPulse_seg0ro
	3935710	0x1a2c9062000	0x1a2c9063000	         4	rw-        	full          	NetPulse_seg1rw
	3935650	0x1a2db63d000	0x1a2db66c000	       188	r-x        	full          	libicutu.so.57.1_seg0ro
	3935651	0x1a2db86c000	0x1a2db86e000	         8	rw-        	full          	libicutu.so.57.1_seg1rw
	3935652	0x1a2db86e000	0x1a2db8a0000	       200	rw-        	full          	libicutu.so.57.1_seg2rw
	3935693	0x1cf6b1c6000	0x1cf6b1cb000	        20	r-x        	full          	libbsd.so_seg0ro
	3935694	0x1cf6b3ca000	0x1cf6b3cb000	         4	rw-        	full          	libbsd.so_seg1rw
	3935661	0x1f596d7d000	0x1f596e79000	      1008	r-x        	full          	libtextencoding.so_seg0ro
	3935662	0x1f597079000	0x1f59707c000	        12	rw-        	full          	libtextencoding.so_seg1rw
	3935646	0x1f6e6a28000	0x1f6e6a7d000	       340	r-x        	full          	libicule.so.57.1_seg0ro
	3935647	0x1f6e6c7d000	0x1f6e6c80000	        12	rw-        	full          	libicule.so.57.1_seg1rw
	3935706	0x7f2fd5865000	0x7f2fd58aa000	       276	rw-s       	full          	_BMediaRoster__45605_stack
	3972054	0x7f461135b000	0x7f46113a0000	       276	rw-s       	full          	vmw cleanup_46495_stack
	3935714	0x7f4720447000	0x7f472048c000	       276	rw-s       	full          	Expando Window Watcher_45606_st
	3935619	0x7f5db6e6b000	0x7f5db6e73000	        32	r-xSk      	full          	commpage
	3935722	0x7f6a9c74f000	0x7f6a9c794000	       276	rw-s       	full          	w>Deskbar_45610_stack
	3935616	0x7f8558929000	0x7f855892d000	        16	rw-        	full          	user area
	3935720	0x7fef0a548000	0x7fef0a58d000	       276	rw-s       	full          	w>Twitcher_45609_stack
	3935618	0x7ffb8e3ce000	0x7ffb8f3d0000	     16392	rw-s       	full          	Deskbar_45600_stack

Protection Flags: r - read, w - write, x - execute, s - stack, o - overcommit, c - cloneable, S - shared, k - kernel

Semaphores:
	ID		Count	Last Holder	Name
	------------------------------------------------------------
	3223310	    0	     -41589	BMessageQueue Lock
	3223318	    0	      41589	expando monitor
	3223319	    0	      41589	some BLocker
	3228736	    0	     -30572	BMessageQueue Lock
	3228763	    0	     -37350	offscreen bitmap
	3229363	    0	     -41589	_BMediaRoster_
	3229369	    0	     -41589	media theme lock
	3229370	    0	     -41589	Tracker shared icon cache
	3229371	    1	     -45610	Deskbar
	3229534	    0	     -41589	some BLocker
	3229544	    0	     -41589	some BLocker
	3229575	    0	     -41589	some BLocker
	3229576	    0	      15328	some BLocker
	3229577	    0	      45606	AppServerLink_sLock
	3229611	    0	      41589	Catalog
	3229684	    0	      45610	token space
	3229686	    0	      46494	BLooperList lock
	3229687	    0	     -41589	PeriodicUpdatePoses
	3229688	    0	     -41589	Light Lock
	3229695	    0	     -25509	LocaleRosterData
	3229696	    0	      45733	some BBlockCache lock
	3229697	    0	     -41589	BShelf loaded image map
	3229699	    0	     -41589	Catalog
	3229703	    0	      41589	Catalog
	3229705	    0	     -41589	media plugin manager
	3229707	    0	     -41589	some BLocker
	3229708	    0	     -41589	some BLocker
	3229709	    0	      41589	some BLocker
	3229710	    0	      41589	Tracker node icon cache
	3229711	    0	     -41589	undo
	3229712	    0	      45610	BMessageQueue Lock
	3229714	    0	      41589	Light Lock
	3229719	    0	      30681	screen list
	3229723	    0	     -41589	AppLooperPort
	3229729	    0	      37474	width buffer
	3229730	    0	     -15328	Light Lock
	3229731	    0	     -15328	clipboard
	3229735	    0	     -41589	tool tip manager
	3229740	    0	     -41589	BMessageQueue Lock
	3229744	    0	      15328	dormant node manager locker
	3229746	    0	     -41589	time source object manager
	3229755	    0	     -41589	some BLocker
	3229758	    0	     -15328	DNS cache
	3229759	    0	      41589	BMediaRoster::Roster locker
	3229761	    0	      41589	synchronized hash map
	3229762	    0	      41589	synchronized hash map
	3229763	    0	     -41589	some BLocker
	3229764	    0	      15328	Catalog
	3229768	    0	     -41589	BMessageQueue Lock
	3229771	    0	     -41589	add-on manager
	3229772	    0	      41589	port pool
	3229773	    0	      41589	shared buffer list
	3229777	    0	     -37497	Catalog
	3229784	    0	     -41589	Catalog
	3229795	    1	     -41589	main_monitor
	3229803	    0	      45609	Twitcher
	3230633	    0	      46495	vmware backdoor lock
	3230704	    0	     -15328	clipboard
	3264020	    0	       1873	BDragger static
	3264119	    0	      -1873	virtual directory manager
```

**Comment by volo-droid** [comment]
@diversys this one also has been fixed by fa0d7fb8dd6b4b864eaa3ab61889f5929be701f4

**Comment by diversys** [comment]
Great!
